### ISSUE TITLE
Bug: [restrict-template-expressions] allowArrays errors on "as const" arrays.

### ISSUE THREAD

**Issue by DanKaplanSES** [typescript-eslint/typescript-eslint#9491]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=MYewdgzgLgBAhgJwOYwLwwNoHIBmIRYA0MWARolgLrwQyiRQDcAUPdDALYRICMaMAA0Qp0AEgDewgL4DGQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu-RAW2LwAhn3oAPYh2TJKAeyaoMAbXDYciaNDnRIAGjXqs67JGHx4cgO4BBLcMLowXWIkPYAvu4C6arx6A&tsconfig=&tokens=false

### Repro Code

```TypeScript
const arg = ['foo', 'bar'] as const;
const msg1 = `arg = ${arg}`;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/restrict-template-expressions": [
      "error",
      {
        "allowArray": true
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected no typescript-eslint error because I set `allowArrays: true` as an option.

### Actual Result

I get an error:

> Invalid type "readonly ["foo", "bar"]" of template literal expression.ESLint [@typescript-eslint/restrict-template-expressions](https://typescript-eslint.io/rules/restrict-template-expressions)

**Comment by auvred** [comment]
Yep, the current implementation of the rule only catches arrays, not tuples!
